sound/soc/sof/sof-pci-dev.c

```diff
@@ -116,14 +116,11 @@ int sof_pci_probe(struct pci_dev *pci, const struct pci_device_id *pci_id)
 	const struct sof_dev_desc *desc =
 		(const struct sof_dev_desc *)pci_id->driver_data;
 	struct snd_sof_pdata *sof_pdata;
-	const struct snd_sof_dsp_ops *ops;
 	int ret;
 
 	dev_dbg(&pci->dev, "PCI DSP detected");
 
-	/* get ops for platform */
-	ops = desc->ops;
-	if (!ops) {
+	if (!desc->ops) {
```

Comment:
this looks good @ujfalusi but I have one more suggestion for a future cleanup. line 141 is redndant as desc is already set. no?

Reply:
I don't think so, we have to take the PCI desc info and pass it to the machine driver, that's a different structure.

Reply:
ahh yes, so it is simply: sof_pdata->desc = desc

Reply:
we could indeed avoid re-casting the pci_id stuff, I think this comes from historical code being reworked and there's no consistency.